vm/src/macros.rs

```diff
@@ -147,3 +147,90 @@ macro_rules! extend_class {
         )*
     }
 }
+
+/// Macro to match on the built-in class of a Python object.
+///
+/// Like `match`, `match_object!` is an expression, and so must have a default
+/// arm that will be evaluated when the object was not an instance of any of
+/// the classes specified in other arms.
```

Comment:
Also, is there any way to enable a missing default path? `match` is allowed as just a statement, where everything just leads to a `()` type. Unless this is to ~sorta emulate algebraic types and requiring that all cases are covered? In which case I'd recommend making the documentation clearer here.

Reply:
Yeah it needs to be exhaustive like `match`, I'll improve the docs